addons/account/models/account_payment.py

```diff
@@ -416,7 +416,7 @@ def _compute_state(self):
                 liquidity, _counterpart, _writeoff = payment._seek_for_lines()
                 payment.state = (
                     'paid'
-                    if move.currency_id.is_zero(sum(liquidity.mapped('amount_residual'))) else
+                    if move.currency_id.is_zero(sum(liquidity.mapped('amount_residual'))) or not liquidity.account_id.reconcile else
```

Comment:
Fun part, there is another bug here:
```suggestion
                    if move.currency_id.is_zero(sum(liquidity.mapped('amount_residual_currency'))) or not liquidity.account_id.reconcile else
```
or
```suggestion
                    if move.company_currency_id.is_zero(sum(liquidity.mapped('amount_residual'))) or not liquidity.account_id.reconcile else
```